qetpy/core/didv/_didv.py

```diff
@@ -1020,63 +1013,71 @@ def calc_bias_params_infinite_loop_gain(self, poles=3,
+            return
+        
+        # loop poles
+        for model_poles in model_list:
             
-        elif  poles == 3:
-            if self._3poleresult is None:
-                raise ValueError(
-                    'ERROR: The 2-pole fit has not been run'
-                )
-            results = self._3poleresult
-
-
-        # check if ibias available
-        if tes_bias is None:
+            results = copy.deepcopy(self._fit_results[model_poles])
+            
+            # check if ibias available
+            if tes_bias is None:
 
-            if ('biasparams' not in results.keys()
-                or 'ibias' not in results['biasparams']):
-                raise ValueError(
-                    'ERROR: Unable to find tes bias (ibias)!'
-                    ' It needs to be provided.'
-                )
+                if ('biasparams' not in results.keys()
+                    or 'ibias' not in results['biasparams']):
+                    raise ValueError(
+                        'ERROR: Unable to find tes bias (ibias)!'
+                        ' It needs to be provided.'
+                    )
             
-            tes_bias = results['biasparams']['ibias']
-            if 'ibias_err' in results['biasparams']:
-                tes_bias_err = results['biasparams']['ibias_err']
+                tes_bias = results['biasparams']['ibias']
+                if 'ibias_err' in results['biasparams']:
+                    tes_bias_err = results['biasparams']['ibias_err']
           
 
-        # Rp
-        if rp is None:
-            if self._rp is None:
-                raise ValueError(
-                    'ERROR: Unable to find rp!'
-                    ' It needs to be provided.'
-                )
-            rp = self._rp
+            # Rp
+            if rp is None:
+                if self._rp is None:
+                    raise ValueError(
+                        'ERROR: Unable to find rp!'
+                        ' It needs to be provided.'
+                    )
+                rp = self._rp
 
             
-        results['biasparams_infinite_lgain'] = (
-            get_tes_bias_parameters_dict_infinite_loop_gain(
-                poles,
-                results['params'], results['cov'],
-                tes_bias, tes_bias_err,
-                self._rsh, rp
+            results['biasparams_infinite_lgain'] = (
+                get_tes_bias_parameters_dict_infinite_loop_gain(
+                    results['params'], results['cov'],
+                    tes_bias, tes_bias_err,
+                    self._rsh, rp
+                )
             )
-        )
-            
-        # replace results
-        if poles == 2:
-            self._2poleresult  = results
-        elif  poles == 3:
-            self._3poleresult = results
-                      
+
+            self._fit_results[model_poles] = results
+                                  
     def dofit_with_true_current(self, offset_dict,
```

Comment:
is this deprecated in the current version of the code? Or should it still be used?

Reply:
Are you referring to dofit_with_true_current function. Yes, I think it should be deprecated to calc_smallsignal_parameters (it should be doing same thing but for both poles).

However, I think we should further cleanup by having the "true current" calculation to be be completely separated.  The "true current" does not even need to be done on dIdV data, we could use randoms.  The code could be separate from dIdV.  The input should a dictionary or pandas Series with I0, I0_err, etc.  I0 could be from IV sweep, true current, or fake muons, or even didv with infinite loop gain. It is a bit too black box at the moment. Separating steps would make things more understandable.  We can always add a funtion dofit_and_calcI0_and calc_ssp(...) if we want...